File: service-common/src/main/java/user/Stat.java
package user;

import data.*;
import lombok.AllArgsConstructor;
import lombok.Data;

import java.util.List;
import java.util.Set;
import java.util.UUID;

@Data
@AllArgsConstructor
public class Stat implements Unique {
    private UUID uuid;

    private String realm;

    private int money;
    private int kills;
    private int wins;
    private int games;
    private int lootbox;
    private int lootboxOpenned;

    private List<Donate> donate;
    private Set<String> donates;
    private List<Achievement> achievement;

    private KillMessage activeKillMessage;
    private StepParticle activeParticle;
    private NameTag activeNameTag;
    private Corpse activeCorpse;
    private StarterKit activeKit;

    private Long timePlayedTotal;

    private Boolean isApprovedResourcepack;
}

File: commons/src/main/kotlin/me/reidj/bridgebuilders/user/Stat.kt
package me.reidj.bridgebuilders.user

import me.func.protocol.battlepass.BattlePassUserData
import me.reidj.bridgebuilders.achievement.Achievement
import me.reidj.bridgebuilders.battlepass.quest.BattlePassQuest
import me.reidj.bridgebuilders.battlepass.quest.QuestGenerator
import me.reidj.bridgebuilders.donate.DonatePosition
import me.reidj.bridgebuilders.donate.impl.*
import java.util.*

data class Stat(
    val id: UUID,

    var money: Int,
    var kills: Int,
    var wins: Int,
    var games: Int,
    var lootbox: Int,
    var lootboxOpenned: Int,

    var progress: BattlePassUserData? = BattlePassUserData(10, false),
    var data: List<BattlePassQuest>? = QuestGenerator.generate(),
    var lastGenerationTime: Long = System.currentTimeMillis(),

    var achievement: MutableList<Achievement>,

    var donate: MutableList<DonatePosition>,
    var activeKillMessage: KillMessage,
    var activeParticle: StepParticle,
    var activeNameTag: NameTag,
    var activeCorpse: Corpse,
    var arrowParticle: ArrowParticle,
    var activeKit: StarterKit,

    var timePlayedTotal: Long,
    var lastEnter: Long,

    var lastSeenName: String?,

    var claimedRewards: MutableList<Int>? = mutableListOf()
)
